libr/include/r_util/r_assert.h

```diff
@@ -0,0 +1,97 @@
+#ifndef R_ASSERT_H
+#define R_ASSERT_H
+
+#if defined (__GNUC__) && defined (__cplusplus)
+#define R_STRFUNC     ((const char*) (__PRETTY_FUNCTION__))
+#elif defined (__STDC_VERSION__) && __STDC_VERSION__ >= 199901L
+#define R_STRFUNC     ((const char*) (__func__))
+#elif defined (__GNUC__) || (defined(_MSC_VER) && (_MSC_VER > 1300))
+#define R_STRFUNC     ((const char*) (__FUNCTION__))
+#else
+#define R_STRFUNC     ((const char*) ("???"))
```

Comment:
uhm, i think __FUNCTION__ should work everywhere

Reply:
`__func__` works in GCC 4.5 or above, but you'd better enable C99 mode or above.

Reply:
How about `R_FUNC_NAME`